core/chains/evm/config/config.go

```diff
@@ -162,6 +163,11 @@ type BlockHistory interface {
 	TransactionPercentile() uint16
 }
 
+type DAOracle interface {
```

Comment:
can we have DAOracle subtypes, where we can define the concrete DAOracle type in chain toml, and get rid of this hardcoded mapping
```
switch chainType {
	case chaintype.ChainOptimismBedrock, chaintype.ChainKroma, chaintype.ChainScroll, chaintype.ChainMantle:
		l1Oracle, err = NewOpStackL1GasOracle(lggr, ethClient, chainType)
	case chaintype.ChainArbitrum:
		l1Oracle, err = NewArbitrumL1GasOracle(lggr, ethClient)
	case chaintype.ChainZkSync:
		l1Oracle = NewZkSyncL1GasOracle(lggr, ethClient)
	default:
		return nil, fmt.Errorf("received unsupported chaintype %s", chainType)
	}
```
today's existing DAOracle types are ArbitrumOracle, zkSyncOracle, and OPOracle

we want to expose
```
GasOracleAddress() gethcommon.Address
CustomGasPriceAPICalldata() []byte
```
specifically for OPOracle type